src/main/java/org/spongepowered/common/mixin/core/entity/passive/MixinEntityVillager.java

```diff
@@ -237,4 +248,25 @@ public void supplyVanillaManipulators(List<DataManipulator<?, ?>> manipulators)
     public Optional<Villager> getCarrier() {
         return Optional.of(this);
     }
+
+    @Redirect(method = "updateEquipmentIfNeeded",  at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/passive/EntityVillager;canVillagerPickupItem(Lnet/minecraft/item/Item;)Z"))
+    public boolean onCanVillagerPickUpItem(EntityVillager villager, Item item, EntityItem itemEntity) {
+        boolean result = this.canVillagerPickupItem(item);
+        if (!SpongeCommonEventFactory.callChangeInventoryPickupPreEvent(((EntityLiving)(Object) this), itemEntity)) {
```

Comment:
This should only be fired if `canVillagerPickupItem()` returns `true`.

Reply:
Likewise, please javadoc what this method was originally doing if possible for future maintenance (as may be noticed in #1882)